core/markets/retrieval/client/impl/retrieval_client_impl.cpp

```diff
@@ -150,7 +211,82 @@ namespace fc::markets::retrieval::client {
   }
 
   void RetrievalClientImpl::processPaymentRequest(
-      const std::shared_ptr<DealState> &deal_state) {}
+      const std::shared_ptr<DealState> &deal_state,
+      const TokenAmount &payment_requested,
+      bool last_payment) {
+    // check that requested is not exceed total funds for deal
+    if (deal_state->funds_spent + payment_requested > deal_state->total_funds) {
+      failDeal(deal_state, RetrievalClientError::kRequestedTooMuch);
+      return;
+    }
+    // not enough bytes received between payment request
+    if ((deal_state->total_received - deal_state->bytes_paid_for
+         < deal_state->current_interval)
+        && !last_payment) {
+      failDeal(deal_state,
+               RetrievalClientError::kBadPaymentRequestBytesNotReceived);
+      return;
+    }
+    // check if payment requested > price of bytes received
+    if (payment_requested
+        > (deal_state->total_received - deal_state->bytes_paid_for)
+              * deal_state->proposal.params.price_per_byte) {
```

Comment:
Here can be a problem with precedence
https://en.cppreference.com/w/cpp/language/operator_precedence

Reply:
Multiplication has higher priority, but I added braces.